.circleci/config.yml

```diff
@@ -0,0 +1,274 @@
+      - run:
+         name: Deprecated code check
+         command: |
+            cd /home/circleci/project
+            vendor/bin/drupal-check /home/circleci/project/web/modules/custom
+
+  # Run any unit tests and any kernel tests against a vanilla D8 site + our site config imported over it (no predefined content).
+  unit_kernel_tests:
+    docker:
+      - image: circleci/php:7.2.20-apache-browsers
+      - image: circleci/mysql:5.7.27
+    steps:
+      # Add SSH user key so we can access related repositories as part of our initial clone + composer install command.
+      - add_ssh_keys:
+          fingerprints:
+            # TODO: change this SSH key fingerprint to dof-dss key, once it is available.
+            - "40:ee:b5:e2:37:3d:e5:e0:79:7f:95:a3:1d:b5:7d:c9"
+      - attach_workspace:
+          at: ./
+      - run:
+            name: Add OS and PHP extensions/config
+            command: |
+              sudo cp /home/circleci/project/.circleci/docker-php-circleci.ini /usr/local/etc/php/conf.d/
+              sudo apt-get update
+              sudo apt-get install -y libpng-dev mariadb-client rsync
+              sudo docker-php-ext-install gd pdo_mysql
+              composer global require hirak/prestissimo
+      - run:
+            name: Copy vhost into place
+            command: |
+              sudo cp .circleci/drupal.vhost /etc/apache2/sites-available/drupal.conf
+      - run:
+            name: Enable web server and vhosts
+            command: |
+              sudo a2enmod rewrite
+              sudo a2dissite 000-default
+              sudo a2ensite drupal
+              sudo service apache2 start
+      - run:
+            name: Install Drupal and configure settings
+            command: |
+              # Should be scaffolded by now.
+              cd /home/circleci/project/web
+              cp sites/default/default.settings.php sites/default/settings.php
+              cp sites/default/default.services.yml sites/default/services.yml
+              # Copy in our environment specific settings to the settings.php file.
+              cp /home/circleci/project/.circleci/drupal.settings.php sites/default/settings.php
+              # Copy PHPUnit config into core folder.
+              cp /home/circleci/project/.circleci/phpunit.circleci.xml core/
+              # Install Drupal (or restore from DB dump at this point).
+              ../vendor/bin/drush site-install -y standard --site-name="NI Direct"
+              # Enable migrate_nidirect_utils.
+              ../vendor/bin/drush en -y migrate_nidirect_utils
+              # No migrate db, so fudge the drupal-console commands into drush equivalents.
+              ../vendor/bin/drush cset -y "system.site" uuid "${UUID}"
+              ../vendor/bin/drush ev '\Drupal::entityManager()->getStorage("shortcut_set")->load("default")->delete();'
```

Comment:
This will blow up. Drupal will complain if the default shortcut set is missing. You need to load each link belonging to that set and delete each of those.
See: https://github.com/dof-dss/nidirect-d8-mig-mods/blob/development/migrate_nidirect_utils/src/Command/NidirectMigratePreCommand.php#L73

Reply:
You can probably tell from the comment I tried to use drupal console to run this, but because I was missing the migrate DB config it didn't want to play any further. I quite like the idea of re-using the code for that command, so I'm tempted to bash in some placeholder values for the config it needs and see how it behaves. That said, those two commands where what I've run before we had the drupal console commands and didn't seem to cause any trouble... but perhaps I've missed a step after that and totally not seen a bigger problem :)

Reply:
I'd originally deleted the default shortcut set thinking it would be ok but once you start trying to use the site it'll throw up a routing error.

Reply:
Okay, so I can't use drupal console directly as I'm missing a bunch of stuff we'd define in Lando. Even if I fake it with placeholder values the command I want to use starts wanting to fudge source db URL aliases and all the other helpful things that are going in when it comes to local dev work.

What I've done is set up a small script I can run in isolation with drush that borrows the bits of code from that command that we need. So the outcome is the same, it safely removes shortcut links, at the tiny expense of not being as DRY and using the command itself via drupal console. A bit irritating, but I can live with that for now. I guess we can create a CI helper command in future if that's more appropriate but I don't see it being a critical issue.